src/mame/drivers/zr107.cpp

```diff
@@ -614,6 +606,15 @@ INPUT_PORTS_END
 static INPUT_PORTS_START( midnrun )
 	PORT_INCLUDE( zr107 )
 
+	PORT_START("IN0")
+	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_START1 ) PORT_NAME("Start/View")
+	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_JOYSTICK_UP ) PORT_NAME("Shift Up")
+	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_JOYSTICK_DOWN ) PORT_NAME("Shift Down")
+	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_BUTTON5 ) PORT_NAME("AT/MT Switch") PORT_TOGGLE PORT_CONDITION("IN3", 0x02, NOTEQUALS, 0x00)
+	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_BUTTON5 ) PORT_NAME("AT/MT Switch") PORT_CONDITION("IN3", 0x02, EQUALS, 0x00)
```

Comment:
Will this cause a problem with configuration saving/loading?  The entries in `cfg` files are identified by (port tag, mask, type) tuple.  You’ve got two controls with the same combination, where one is a toggle and one isn’t.  Have you checked what happens when you change DIP switch settings and change the input mapping for this?

Reply:
No issues presented itself when testing including saving/loading.  If you change the DIP switch setting and reset the machine or exit MAME and load the game back up, the DIP switch setting is retained correctly and the game is working with the stored DIP switch setting.  I have carried this test out many times with Winding Heat and Midnight Run successfully.

Reply:
I wasn’t talking about saving the *DIP switch* setting, I was talking about saving *input assignments* for the two buttons with identical config tuples.

But thinking about this further, for the T-gate shifter, you can’t activate shift up/down and auto at the same time.  Should it be something like:

```
 	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_JOYSTICK_UP ) PORT_NAME("Shift Up") PORT_4WAY
	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_JOYSTICK_DOWN ) PORT_NAME("Shift Down") PORT_4WAY
	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_JOYSTICK_LEFT ) PORT_NAME("Auto Shift") PORT_4WAY PORT_TOGGLE PORT_CONDITION("IN3", 0x02, NOTEQUALS, 0x00)
	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_BUTTON5 ) PORT_NAME("AT/MT Switch") PORT_CONDITION("IN3", 0x02, EQUALS, 0x00)
```

Reply:
Thanks for the response.  Now I understand what you mean about the identical config tuples.  That can be easily corrected and the saving of input assignments is affected with them being the config tuples being the same.

I did actually try something that was almost identical to what you suggested above but it didn't have the desired effect.  It didn't prevent Winding Heat from seeing the Shift Up and Shift Down with Auto mode selected in 'T'Gate shifter mode. 
Refer to commit 3943022f0f0b1a4b2793287735114b8856c17774.

This is the code I had created in the above commit:
	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_JOYSTICK_UP ) PORT_NAME("Shift Up") PORT_4WAY
	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_JOYSTICK_DOWN ) PORT_NAME("Shift Down") PORT_4WAY
	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_JOYSTICK_LEFT ) PORT_NAME("Auto Transmission") PORT_4WAY PORT_TOGGLE PORT_CONDITION("IN3", 0x02, NOTEQUALS, 0x00)
	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_BUTTON5 ) PORT_NAME("AT/MT Switch") PORT_CONDITION("IN3", 0x02, EQUALS, 0x00)

Reply:
Well, not handling toggles in joystick inputs as well as it should is probably a limitation of the input system right now.  It still better to do it as a 4-way joystick with no right direction for documentation purposes.

Reply:
Agreed.  I will update it in the next 5 minutes or so.  I have just been looking at the way 4 way joysticks are handled in the ioport.cpp and if the joystick is moved left then moved up or down, the up and down input is allowed.  The following comment in the code explains the reason for this:

		//  If joystick is pointing at a diagonal, acknowledge that the player moved
		//  the joystick by favoring a direction change.  This minimizes frustration
		//  and maximizes responsiveness.
		//
		//  For example, if you are holding "left" then switch to "up" (where both left
		//  and up are briefly pressed at the same time), we'll transition immediately
		//  to "up."

I plan to add a new macro to support the 'T'Gate.  I will do this in a separate PR

Reply:
It doesn’t need another macro.

Reply:
Ok noted.